core/src/main/java/hudson/Launcher.java

```diff
@@ -441,6 +447,22 @@ public ProcStarter writeStdin() {
             return this;
         }
 
+        /**
+         * Indicates that the launched process should not be killed when interrupted.
+         * It allows detecting the interruption on caller's side and do custom (cleanup) action while
+         * the launched process is still running.
+         *
+         * <p>
+         * Note that the process can (and should) be killed
+         * via {@link Proc#kill()} when custom action is done.
```

Comment:
This indicates to me that the better API design is to submit a callback to be invoked between interrupting and killing, rather than just a flag to skip it.

Reply:
I agree. But it makes the implementation more complicated...
@oleg-nenashev could you recommend a way to pass the callback? I tried to pass hudson.remoting.Callable but I got NotSerializableExceptions.